### PR TITLE
Stochastic gradient descent

### PR DESCRIPTION
**Pull request by bheim6**
@jmejia We are now pulling in batches of data to train the net. We are also now using data representative of data in our db. This PR also contains the moving of Neural Net PORO into lib/.

### CODE REVIEW

**Comment by JeanJoeris** [comment]
Sweet! We have the algorithm pretty much working
